adyenofficial.php

```diff
@@ -325,6 +328,63 @@ private function createAdyenDatabaseTables()
             $this->createAdyenNotificationTable();
     }
 
+    /**
+     * Copy waiting_for_payment email template to all /mails/ subdirectories
+     * Required until https://github.com/PrestaShop/PrestaShop/issues/24336 is implemented
+     *
+     * @return bool
+     */
+    public function copyEmailTemplates()
+    {
+        $allDirectories = true;
+        $mailsDirectory = _PS_ROOT_DIR_.'/mails/';
+        $adyenEmailDirectory = __DIR__ . '/views/templates/email/';
+        if ($handle = opendir($mailsDirectory)) {
+            while (false !== ($entry = readdir($handle))) {
+                $languageDirectory = $mailsDirectory . $entry;
+                if (is_dir($languageDirectory)) {
+                    $adyenHtmlFile = $adyenEmailDirectory . 'waiting_for_payment_adyen.html';
+                    $adyenTxtFile = $adyenEmailDirectory . 'waiting_for_payment_adyen.txt';
+                    if (!copy($adyenHtmlFile, $languageDirectory . '/waiting_for_payment_adyen.html') ||
+                        !copy($adyenTxtFile, $languageDirectory . '/waiting_for_payment_adyen.txt')
+                    ) {
+                        $this->logger->error(
+                            sprintf('Unable to copy email template to directory: %s', $languageDirectory)
+                        );
+                        $allDirectories = false;
+                    }
+                }
+            }
+        }
+
+        return $allDirectories;
+    }
+
+    /**
+     * Remove waiting_for_payment email template from all /mails/ subdirectories
+     *
+     * @return bool
+     */
+    private function removeCopiedEmailTemplates()
+    {
+        $mailsDirectory = _PS_ROOT_DIR_.'/mails/';
+        if ($handle = opendir($mailsDirectory)) {
+            while (false !== ($entry = readdir($handle))) {
+                $languageDirectory = $mailsDirectory . $entry;
+                if (is_dir($languageDirectory)) {
+                    if (file_exists($languageDirectory . '/waiting_for_payment_adyen.html')) {
+                        unlink($languageDirectory . '/waiting_for_payment_adyen.html');
```

Comment:
Just to be a bit more on the defensive and to make sure if the file delete didn't go well we log it. If you find it useful please do it in the line 379 as well.

```suggestion
                        if (!unlink($languageDirectory . '/waiting_for_payment_adyen.html')) {
                            $this->logger->error(
                                sprintf('Unable to delete email template from directory: %s', $languageDirectory)
                            );
                        }
```

Reply:
https://github.com/Adyen/adyen-prestashop/pull/195/commits/8c34dde4a8600fc3e1ced7944cd81bd8517a9247